inbox/quick-response.xml

```diff
@@ -0,0 +1,212 @@
+    <p>Clients that receive a message containing allowed responses can offer UI to quickly and conveniently select one of the allowed responses.</p>
+  </section2>
+
+  <section2 anchor="selecting-a-response" topic="Selecting a Response">
+    <p>When the user selects a response, their client sends a plaintext message body containing the <tt>@value</tt> as <tt>&lt;body&gt;</tt> text. The client uses the <tt>@value</tt> that corresponds to the <tt>&lt;translation&gt;</tt> of the same language as the <tt>&lt;body&gt;</tt>.</p>
+
+<example><![CDATA[<message to="rootbot@example.com">
+  <body xml:lang="en">no</body>
+</message>]]></example>
+
+    <p>The sender of the original message, in this example <tt>rootbot@example.com</tt>, checks incoming messages for a <tt>&lt;body&gt;</tt> that only contains the <tt>@value</tt> of the corresponding <tt>&lt;translation&gt;</tt> to see if a response was selected. In this example, the <tt>&lt;body&gt;</tt> matches the <tt>@value</tt> of the english translation for the response "No".</p>
+  </section2>
+
+  <section2 anchor="sending-available-actions" topic="Sending a Set of Available Actions">
+    <p>A message with available actions is sent by including one or more &lt;available-action&gt; elements with distinct <tt>@id</tt>s.</p>
+
+<example><![CDATA[<message to="gitbot@example.com">
+  <body xml:lang="en">New merge request opened by ExampleUser: https://git.example.com/example/mrs/3/</body>
+  <available-action xmlns="]]>&ns;<![CDATA[" id="merge">
+    <label xml:lang="en">Merge Now</label>
+  </available-action>
+</message>]]></example>
+
+    <p>A single message MUST NOT contain multiple <tt>&lt;available-action&gt;</tt> elements with the same values for the <tt>@id</tt> attribute. Labels MUST be unique per id and language.</p>
+    <p>Clients that receive a message containing available actions SHOULD offer UI to select one of the actions.</p>
+  </section2>
+
+  <section2 anchor="selecting-an-action" topic="Selecting an Action">
+    <p>When the user selects an action, their client sends a message containing an <tt>&lt;action-selected&gt;</tt> element which identifies the selected action. The message does not contain a <tt>&lt;body&gt;</tt>.</p>
+
+<example><![CDATA[<message to="gitbot@example.com">
+  <action-selected xmlns="]]>&ns;<![CDATA[" id="merge" />
+</message>]]></example>
+  </section2>
+</section1>
+
+<section1 anchor="rules" topic="Business Rules">
+  <p>All message bodies SHOULD always list the (internationalized) allowed responses too, so that users of clients that don't support &xepname; can still know what the allowed responses are.</p>
+  <p>Bots or other entities that indicate allowed responses are free to accept variations of the response values for convenience, for example ignoring the casing or accepting abbreviations of response values.</p>
```

Comment:
This should probably use normative language since it's in the business rules: "MAY accept variations"

Reply:
hmm, I don't agree that this should use normative language, maybe it shouldn't be part of the XEP at all, I'm really unsure. I'll move it to implementation notes for now and then I'll wait for further feedback.